code/datums/components/tackle.dm

```diff
@@ -0,0 +1,485 @@
+  *	the roll, the more severe the result.
+  *
+  * Mod 1: Speed
+  *	* Base tackle speed is 1, which is what normal gripper gloves use. For other sources with higher speed tackles, like dolphin and ESPECIALLY rocket gloves, we obey Newton's laws and hit things harder.
+  *	* For every unit of speed above 1, move the lower bound of the roll up by 15. Unlike Mod 2, this only serves to raise the lower bound, so it can't be directly counteracted by anything you can control.
+  *
+  * Mod 2: Misc
+  *	-Flat modifiers, these take whatever you rolled and add/subtract to it, with the end result capped between the minimum from Mod 1 and 100. Note that since we can't roll higher than 100 to start with,
+  *		wearing a helmet should be enough to remove any chance of permanently paralyzing yourself and dramatically lessen knocking yourself unconscious, even with rocket gloves. Will expand on maybe
+  *	* Wearing a helmet: -6
+  *	* Wearing riot armor: -6
+  *	* Clumsy: +6
+  *
+  * Effects: Below are the outcomes based off your roll, in order of increasing severity
+  *	* 1-63: Knocked down for a few seconds and a bit of brute and stamina damage
+  *	* 64-83: Knocked silly, gain some confusion as well as the above
+  *	* 84-93: Cranial trauma, get a concussion and more confusion, plus more damage
+  *	* 94-98: Knocked unconscious, significant chance to get a random mild brain trauma, as well as a fair amount of damage
+  *	* 99-100: Break your spinal cord, get paralyzed, take a bunch of damage too. Very unlucky!
+*/
+/datum/component/tackler/proc/splat(mob/living/carbon/user, atom/hit)
+	if(istype(hit, /obj/machinery/vending)) // before we do anything else-
+		var/obj/machinery/vending/darth_vendor = hit
+		darth_vendor.tilt(user, TRUE)
+		return
+	else if(istype(hit, /obj/structure/window))
+		var/obj/structure/window/W = hit
+		splatWindow(user, W)
+		if(QDELETED(W))
+			return COMPONENT_MOVABLE_IMPACT_NEVERMIND
+		return
+
+	var/oopsie_mod = 0
+	var/danger_zone = (speed - 1) * 15 // for every extra speed we have over 1, take away 10 of the safest chance
+	danger_zone = max(min(danger_zone, 100), 1)
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/S = user
+		var/head_slot = S.get_item_by_slot(ITEM_SLOT_HEAD)
+		var/suit_slot = S.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+		if(head_slot && (istype(head_slot,/obj/item/clothing/head/helmet) || istype(head_slot,/obj/item/clothing/head/hardhat)))
+			oopsie_mod -= 6
+		if(suit_slot && (istype(suit_slot,/obj/item/clothing/suit/armor/riot)))
+			oopsie_mod -= 6
+
+	if(HAS_TRAIT(user, TRAIT_CLUMSY))
+		oopsie_mod += 6 //honk!
+
+	var/oopsie = rand(danger_zone, 100)
+	if(oopsie >= 94 && oopsie_mod < 0) // good job avoiding getting paralyzed! gold star!
+		to_chat(user, "<span class='usernotice'>You're really glad you're wearing protection!</span>")
+	oopsie += oopsie_mod
+
+	switch(oopsie)
+		if(99 to 100)
```

Comment:
this should be 99 to infinity, in case a clown rolls a 100 for their oopsie value and then adds 6 to that to get 106